### ISSUE TITLE
Bug with alias as implicit dependency

### ISSUE THREAD

**Issue by gorunovanton** [ninja-build/ninja#1153]
If an alias to a target is used as an implicit dependency, the target won't be rebuilt when it's dependency gets rebuilt.
Test case:
1. Shared library depends on static library;
2. Build all targets;
3. Change static library and build it;
4. Next try to build shared library. Ninja says "ninja: no work to do.".

Code for this case can be found [here](http://pastebin.com/eeJ8UGnq)

If a static library is used as an implicit dependency ($outdir/somelib.lib) in a shared library's build rule, all works perfectly.

**Comment by nico** [comment]
Thanks for the report! I think this is a dupe of #478 and fixed by #656.

**Comment by gorunovanton** [comment]
Yes, it seems like this is the same issue as #478. But it was fixed likely by #1021 instead of #656.

**Comment by nico** [comment]
Yes, sorry, copy-paste error.
